antelope_contracts/contracts/erc20/src/erc20.cpp

```diff
@@ -8,92 +8,269 @@
-    call_data.insert(call_data.end(), deploy_addr.bytes, deploy_addr.bytes + kAddressLength);
-    // Assume account opened in evm_runtime
-    call_act.send(get_self(), to, 0, call_data, evm_init_gaslimit);
+    call_data.insert(call_data.end(), impl_address_bytes->begin(), impl_address_bytes->end());
 
-    // Assume nonce...
-    deploy_addr = silkworm::create_address(reserved_addr, 1); 
+    bytes constructor_data;
+    // sha(function initialize(uint8 _precision,uint256 _egressFee,string memory _name,string memory _symbol,string memory _eos_token_contract)) == 0xd66d4ac3
+    uint8_t func_[4] = {0xd6,0x6d,0x4a,0xc3};
+    constructor_data.insert(constructor_data.end(), func_, func_ + sizeof(func_));
 
-    // TODO: Where can we get the addr...
-    config new_config = {
-        .erc20_addr = {},
+    auto pack_uint256 = [&](bytes &ds, const intx::uint256 &val) {
+        uint8_t val_[32] = {};
+        intx::be::store(val_, val);
+        ds.insert(ds.end(), val_, val_ + sizeof(val_));
     };
+    auto pack_uint32 = [&](bytes &ds, uint32_t val) {
+        uint8_t val_[32] = {};
+        val_[28] = (uint8_t)(val >> 24);
+        val_[29] = (uint8_t)(val >> 16);
+        val_[30] = (uint8_t)(val >> 8);
+        val_[31] = (uint8_t)val;
+        ds.insert(ds.end(), val_, val_ + sizeof(val_));
+    };
+    auto pack_string = [&](bytes &ds, const auto &str) {
+        pack_uint32(ds, (uint32_t)str.size());
+        for (size_t i = 0; i < (str.size() + 31) / 32 * 32; i += 32) {
+            uint8_t name_[32] = {};
+            memcpy(name_, str.data() + i, i + 32 > str.size() ? str.size() - i : 32);
+            ds.insert(ds.end(), name_, name_ + sizeof(name_));
+        }
+    };
+
+    pack_uint32(constructor_data, (uint8_t)erc20_precision); // offset 0
+    pack_uint256(constructor_data, egress_fee_evm);          // offset 32
+    pack_uint32(constructor_data, 160);                      // offset 64
+    pack_uint32(constructor_data, 224);                      // offset 96
+    pack_uint32(constructor_data, 288);                      // offset 128
+    pack_string(constructor_data, evm_token_name);           // offset 160
+    pack_string(constructor_data, evm_token_symbol);         // offset 224
+    pack_string(constructor_data, eos_contract_name.to_string()); // offset 288
 
-    new_config.erc20_addr.resize(kAddressLength);
-    memcpy(new_config.erc20_addr.data(), deploy_addr.bytes, kAddressLength);
+    pack_uint32(call_data, 64);                  // offset 32
+    pack_string(call_data, constructor_data);    // offset 64
 
-    _config.set(new_config, get_self());
+    bytes to = {};
+    bytes value_zero; 
+    value_zero.resize(32, 0);
+
+     // required account opened in evm_runtime
+    call_action call_act(evm_account, {{get_self(), "active"_n}});
+    call_act.send(get_self(), to, value_zero, call_data, evm_init_gaslimit);
+
+    evmc::address proxy_contract_addr = silkworm::create_address(reserved_addr, my_next_nonce()); 
```

Comment:
Send inline `assertnonce` action to ensure your retrieved nonce used to calculate the `proxy_contract_addr` is correct.

Reply:
fixed, assertnonce will be sent in get_next_nonce()